CI Workflow: Add all Azure supported k8s versions

brlbil opened the pull request:
The current aks conformance tests run against one version of k8s either a specific version or the default version that is provided by the cloud provider.

This PR adds all the supported k8s versions by Azure in a matrix strategy.

Other cloud providers will be added by subsequent PRs.

Normal run examples:
:white_check_mark: [conformance-aks-v1.12.yaml](https://github.com/cilium/cilium/actions/runs/5313019853)
:white_check_mark: [conformance-aks-v1.13.yaml](https://github.com/cilium/cilium/actions/runs/5313019842)
:white_check_mark: [conformance-aks.yaml](https://github.com/cilium/cilium/actions/runs/5313019860)

Schedule run simulated:
:white_check_mark: [conformance-aks-v1.12.yaml](https://github.com/cilium/cilium/actions/runs/5313506446)
:x: [conformance-aks-v1.13.yaml](https://github.com/cilium/cilium/actions/runs/5314131467) ,  failures are not related cluster installation
:white_check_mark: [conformance-aks.yaml](https://github.com/cilium/cilium/actions/runs/5313506456)

nbusseneau commented on .github/workflows/lint-workflows.yaml:
```diff
@@ -157,3 +157,30 @@ jobs:
+              fi
+            done
+          done
+
```
Ideally this would be handled in a separate PR on which all of #26356, #26351 and #26354 depend, but it's OK to just have the other PRs depend on this one as well. I think at the very least we should split this in a separate commit though.

brlbil replied:
It was good that this part was highlighted because tests did not cover some of the files.
Fixed it, and separated it as a commit.

nickolaev commented on .github/workflows/conformance-aks-v1.11.yaml:
```diff
@@ -114,11 +112,103 @@ jobs:
               - '!(test|Documentation)/**'
 
+  setup-report:
+    runs-on: ubuntu-latest
```
Do we always use latest? Shall we align to LTS (22.04)?

brlbil replied:
Yes, all CI uses this

brlbil committed:
ci/workflow: add all azure supported k8s versions

The current aks conformance tests run against one version of k8s
either a specific version or the default version that is provided
by the cloud provider.

This commit adds all the supported k8s versions by Azure
in a matrix strategy.

Signed-off-by: Birol Bilgin <birol@cilium.io>

brlbil committed:
ci/workflow: add schema validation

This commit adds schema validation for conformance
tests action files

Signed-off-by: Birol Bilgin <birol@cilium.io>

bimmlerd commented on .github/actions/azure/k8s-versions-schema.yaml:
```diff
@@ -0,0 +1,6 @@
+include: list(include('includeItem'))
```
somewhat meta, but should `.github/actions/` (maybe even all of `.github`) be CODEOWNED by by the `cilium/ci-structure` team instead of contributing? This feels far in the GHA action territory.

bimmlerd reviewed (approved):
I don't feel particularly qualified, see comment, but what is owned by contributing LGTM.

tgraf commented on .github/actions/azure/k8s-versions.yaml:
```diff
@@ -0,0 +1,10 @@
+# List of k8s version for AKS tests
+---
+include:
+  - version: "1.24"
```
Are we unintentionally removing 1.23 test coverage here? The commit message says that this adds "all supported versions" so I'm not sure if the removal of 1.23 is intentional.

nbusseneau replied:
1.23 is not supported on Azure anymore, so we removed it as part of our usual cycling (otherwise what happens the installation will fail consistently due the K8s version being refused by the Azure API on cluster creation) => https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli

tgraf replied:
OK, thanks. Not clear from the commit message.

brlbil replied:
This table was created from the supported k8s for AKS, https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli#aks-kubernetes-release-calendar
